### PR TITLE
Use Scons in CI with container

### PR DESCRIPTION
**Pull request by ntindle**

### CODE REVIEW

**Comment by ntindle** [comment]
Other than the fact that the container doesn't have cargo it works

**Comment by Amaras** [comment]
Alright, let's try and I'll revert if it doesn't run correctly

**Comment by Amaras** [comment]
@ntindle here is the full problem:
Neither `Cargo` nor `rustc` can be found, leading to the build failure (see build 347 I think).

The probable cause is as follows:

1. When the Rust installer runs, it writes the path manipulation inside $HOME/.bashrc
2. The GH actions run through `sh`, which doesn't run the `~/.bashrc` file.
3. Pulling the container, we use `bash` as a shell.

As such, the solution is to do the path manipulation ourselves in the Dockerfile to ensure consistent paths, even with `sh`

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -14,3 +17,10 @@ jobs:
         run: |
           npm install
           npx honkit build
+  
+      - name: check user
+        run: whoami
+      - name: check cargo
+        run: sudo which cargo
+      - name: Run Scons
+        run: sudo scons
```
[the remark this answers is not in the document]
Comment (ntindle) [reply .github/workflows/build.yml]: It also fails without sudo, the sudo was the last thing I tried before going to bed for the night

**Comment by ntindle** [comment]
Okay if whoever merges this doesn't squash and merge, I will cry. THank you for coming to my ted talk

**Review by leios — APPROVED** [review]
Ok, this looks fine for me; however, I am not sure how to test this properly.

I am happy to merge though, I trust your judgement

**Pull request merged**